### ISSUE TITLE
BUG: Cadwyn ignores `generate_unique_id_function` for OpenAPI schema

### ISSUE THREAD

**Issue by isaacharrisholt** [zmievsa/cadwyn#163]
**Describe the bug**

The `generate_unique_id_function` kwarg for `FastAPI` is used in the OpenAPI schema for routing. It doesn't seem to work with Cadwyn.

**Comment by zmievsa** [comment]
Could  you help me out with a reproducible example?

**Comment by isaacharrisholt** [comment]
Sure!

```python
from datetime import date

from cadwyn import Cadwyn
from cadwyn.structure import VersionBundle, Version

from .data import latest

def generate_openapi_id(route: APIRoute) -> str:
    print(route)
    if not route.tags:
        return route.name
    return f'{str(route.tags[0]).lower().replace(' ', '_')}:{route.name}'

versions = VersionBundle(
    Version(date(2024, 3, 5)),
    latest_schemas_package=latest
)

app = Cadwyn(
    versions=versions,
    generate_unique_id_function=generate_openapi_id,
)

@app.get('/', tags=['misc'])
def home():
    return 'Helo, byd!'
```

If you run this and go to the docs URL, you'll notice that the URL for the `/` does not end in `misc:home` like it should.

**Comment by zmievsa** [comment]
@isaacharrisholt  Fixed everything related to this in 3.12.0. Please, check :pray:

**Comment by isaacharrisholt** [comment]
Yep! Seems to work
